### PR TITLE
fix: provider passing in options

### PR DESCRIPTION
**Pull request by prathmeshkhandelwal1**
## Type of change

- [ ] Docs change / dependency upgrade
- [ ] Configuration / tooling changes
- [ ] Refactoring
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires changes in customer code

## Related Issue(s)

#4730

### CODE REVIEW

File: packages/agents/sdk-wrapper/src/sdkPool.ts [hunk 1]
```diff
@@ -39,10 +39,14 @@ export class SdkPool extends SdkShared {
     amount: BigNumberish,
     options?: Options,
   ): Promise<BigNumber> {
-    const _options = options ?? {
-      chains: this.config.chains,
-      signerAddress: this.config.signerAddress,
-    };
+    const originProviderUrl = this.config.chains[domainId]?.providers;
+
+    const _options = options ?? {};
```
Comment (just-a-node) [inline packages/agents/sdk-wrapper/src/sdkPool.ts:44]: We still need to pass chains and signerAddress from config

Comment (prathmeshkhandelwal1) [reply packages/agents/sdk-wrapper/src/sdkPool.ts:44]: So these specific functions dont really require chains and signerAddress internally.

Comment (just-a-node) [reply packages/agents/sdk-wrapper/src/sdkPool.ts:44]: swap() will need signerAddress, for example

File: packages/agents/sdk-wrapper/src/sdkPool.ts [hunk 2]
```diff
@@ -41,8 +41,10 @@ export class SdkPool extends SdkShared {
   ): Promise<BigNumber> {
     const _options = options ?? {
       chains: this.config.chains,
+      originProviderUrl: this.config.chains[domainId]?.providers?.[0],
```
Comment (sanchaymittal) [inline packages/agents/sdk-wrapper/src/sdkPool.ts:44]: We need to use fallback provider here, can't rely on first provider implementation.
Ideally we should be using `chains` options for wrapper-sdk and fetch the provider from it under the core-sdk. single_url options for origin and destination were added for any direct request to sdk-server only.

**Comment by prathmeshkhandelwal1** [comment]
Reverting this to @just-a-node Commit. If we are making changes according to this conversation https://github.com/connext/monorepo/pull/4731#discussion_r1277188024 . Then those features has been added to the PR attached here : #4736 
**We need to merge above mentioned PR before merging this.**

File: packages/agents/sdk-wrapper/src/sdkPool.ts (hunk 2 above)
Comment (just-a-node) [reply packages/agents/sdk-wrapper/src/sdkPool.ts]: Now getting provider from `chains` config that is supplied by client during SDK init.

**Review by just-a-node — APPROVED** [review]

**Review by rhlsthrm — APPROVED** [review]

**Pull request merged**